grafana/grafana
# Logs: Do not insert log-line into log-fields in json download

currently, when you download the logs as JSON, the original log-line will be part of the "fields" section for every log-line (so every log-line line will be there twice).

NOTE: in the tests in `public/app/features/logs/utils.test.ts` i made sure the linked dataframe is more complete, because the tested function will look into that frame to decide what to do.

how to test:
1. open some logs (for example in Loki or Elastic), click the [Download] button, choose `json`. check the downloaded JSON, verify that the log-line does not appear in the `fields` part
2. make sure correlations on the log-line still work:
    - enable the `correlations` feature-flag
    - create a correlation in the `administration/correlations` section. the important part is that in step3, you select the correct datasource, and enter the correct field-name. if you do the test with Loki, the field-name should be `Line`. 
    - now open some logs in explore, expand the log-details on a log-line, and verify that you can see the link created by the correlation
